Advice to build in OSX?

So i tried to compile with mono http://www.go-mono.com/mono-downloads/download.html.
I ran 
xbuild scallion.sln
I got:  https://dl.dropboxusercontent.com/u/39497/xbuild.scallion.log.zip (sorry too big for pastebin)

Then trying to run it:
mono scallion/bin/Debug/scallion.exe -l
I get
`Unhandled Exception:
System.InvalidOperationException: No kernels or only kernel prototypes found when build executable.
  at scallion.CLContext.CreateAndCompileProgram (System.String source) [0x00000] in <filename unknown>:0 
  at scallion.Program.getPreferredWorkGroupSize (IntPtr deviceId) [0x00000] in <filename unknown>:0 
  at scallion.Program.ListDevices () [0x00000] in <filename unknown>:0 
  at scallion.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: No kernels or only kernel prototypes found when build executable.
  at scallion.CLContext.CreateAndCompileProgram (System.String source) [0x00000] in <filename unknown>:0 
  at scallion.Program.getPreferredWorkGroupSize (IntPtr deviceId) [0x00000] in <filename unknown>:0 
  at scallion.Program.ListDevices () [0x00000] in <filename unknown>:0 
  at scallion.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
deweydb:scallion deweydb$ xbuild scallion.sln`

--- a/scallion/Program.cs
+++ b/scallion/Program.cs
@@ -88,7 +88,7 @@ namespace scallion
             ulong preferredWorkGroupSize;
             {
                 CLContext context = new CLContext(deviceId);
-                IntPtr program = context.CreateAndCompileProgram(@"__kernel void get_size() { }");
+                IntPtr program = context.CreateAndCompileProgram(@"__kernel void get_size(__global float2 *in) { }");
                 CLKernel kernel = context.CreateKernel(program, "get_size");
                 preferredWorkGroupSize = kernel.KernelPreferredWorkGroupSizeMultiple;
                 kernel.Dispose();

